### PR TITLE
profile test: allow both the old and the new mangling

### PR DESCRIPTION
**Pull request by rainers**
needed for https://github.com/dlang/dmd/pull/6998

### CODE REVIEW

File: test/profile/Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ $(ROOT)/both.done: $(ROOT)/%.done: $(ROOT)/%
 	$(QUIET)$(TIMELIMIT)$(ROOT)/$* $(ROOT)/both.log $(ROOT)/both.def $(ROOT)/bothgc.log
 	$(QUIET)$(GREP) -q '1 .*_Dmain' $(ROOT)/both.log
 	$(QUIET)$(GREP) -q '1000 .*both.Num\* both.foo(uint)' $(ROOT)/both.log
-	$(QUIET)$(DIFF) both.def.exp $(ROOT)/both.def
+	$(QUIET)$(DIFF) both.def.exp $(ROOT)/both.def || $(DIFF) both.def.exp2 $(ROOT)/both.def
```
Comment (wilzbach) [inline test/profile/Makefile]: Maybe adding a new comment that this allows the old & new mangling and can be removed in the future wouldn't hurt?

File: test/profile/both.def.exp2 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+FUNCTIONS
+	_Dmain
```
Comment (wilzbach) [inline test/profile/both.def.exp2]: You have tabs here ... (you need to either replace them or make the check [more specific](https://github.com/dlang/druntime/blob/master/posix.mak#L279))..

Comment (rainers) [inline test/profile/both.def.exp2]: Those tabs are generated as part of the profile output and are also in both.exp.

File: test/profile/Makefile (hunk 1 above)
Comment (rainers) [inline test/profile/Makefile]: Amended a comment

File: test/profile/both.def.exp2 (hunk 2 above)
Comment (wilzbach) [inline test/profile/both.def.exp2]: I know - I was just FYIing the CI error.
I didn't add the whitespace check here, but it seems that [`.exp` files are explicitly filtered out](https://github.com/dlang/druntime/blob/master/posix.mak#L274). So the simple fix is to rename the file ...

Comment (rainers) [inline test/profile/both.def.exp2]: Sorry, misunderstood at first. Renamed now.

Comment (wilzbach) [inline test/profile/both.def.exp2]: Thanks!

**Pull request merged**